slowmovie.py

```diff
@@ -10,205 +10,233 @@
+    if not os.path.isfile(value):
+        raise argparse.ArgumentTypeError("File '%s' does not exist" % value)
+    if not supported_filetype(value):
+        raise argparse.ArgumentTypeError("'%s' is not a supported file type" % value)
     return value
 
-# Ensure this is the correct path to your video folder 
-viddir = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'Videos/')
-logdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'logs/')
-
-
-parser = argparse.ArgumentParser(description='SlowMovie Settings')
-parser.add_argument('-r', '--random', action='store_true', 
-    help="Random mode: chooses a random frame every refresh")
-parser.add_argument('-f', '--file', type=check_mp4,
-    help="Add a filename to start playing a specific film. Otherwise will pick a random file, and will move to another film randomly afterwards.")
-parser.add_argument('-d', '--delay',  default=120, 
-    help="Delay between screen updates, in seconds")
-parser.add_argument('-i', '--inc',  default=4, 
-    help="Number of frames skipped between screen updates")
-parser.add_argument('-s', '--start',  
-    help="Start at a specific frame")
+def check_dir(value):
+    if os.path.isdir(value):
+        return value
+    else:
+        raise argparse.ArgumentTypeError("Directory '%s' does not exist" % value)
+
+def supported_filetype(file):
+    _, ext = os.path.splitext(file)
+    return ext.lower() in fileTypes
+
+def video_info(file):
+    videoInfo = ffmpeg.probe(file)
+    frameCount = int(videoInfo["streams"][0]["nb_frames"])
+    framerate = videoInfo["streams"][0]["avg_frame_rate"]
+    framerate = float(Fraction(framerate))
+    frametime = 1000 / framerate
+    return frameCount, framerate, frametime
+
+os.chdir(os.path.dirname(os.path.realpath(__file__)))
+
+if "configargparse" in sys.modules:
+    parser = argparse.ArgumentParser(default_config_files=["slowmovie.conf"])
+else:
+    parser = argparse.ArgumentParser()
+parser.add_argument("-D", "--dir", default = "Videos", type = check_dir, help = "Select video directory")
+parser.add_argument("-r", "--random", action = "store_true", help = "Display random frames")
+parser.add_argument("-R", "--random-file", action = "store_true", help = "Play files in random order")
+parser.add_argument("-f", "--file", type = check_mp4, help = "Specify an MP4 file to play")
+parser.add_argument("-d", "--delay", default = timeInterval, type = int, help = "Time between updates, in seconds")
+parser.add_argument("-i", "--increment", default = frameIncrement, type = int, help = "Number of frames to advance on update")
+parser.add_argument("-s", "--start", type = int, help = "Start at a specific frame")
+parser.add_argument("-c", "--contrast", default=contrast, type=float, help = "Adjust image contrast (default: 1.0)")
+parser.add_argument("-a", "--adjust-delay", action = "store_true", help = "Reduce delay by the amount of time taken to display a frame.")
```

Comment:
Why is adjust delay optional? Wouldn't you always want the delay to be as accurate as possible? One less command line argument.

Reply:
I actually don't know why you wouldn't want it. I put that as a 'just-in-case', but I suppose it isn't really needed.